src/idl_parser.cpp

```diff
@@ -853,29 +853,62 @@ void Parser::SerializeStruct(const StructDef &struct_def, const Value &val) {
 
 CheckedError Parser::ParseTable(const StructDef &struct_def, std::string *value,
                                 uoffset_t *ovalue) {
-  EXPECT('{');
+
+  // takes care of nested json array, e.g. [[1,2], [3,4]]
```

Comment:
Generalize comment: "We allow tables both as JSON object {..} with field names or vector [..] with all fields in order." or similar.